src/tools/spin/spinquic.cpp

```diff
@@ -606,37 +613,48 @@ CXPLAT_THREAD_CALLBACK(ServerSpin, Context)
     QuicSettings.IsSet.PeerUnidiStreamCount = TRUE;
     // TODO - Randomize more of the settings.
 
-    ASSERT_ON_FAILURE(
+    auto CredConfig = CxPlatGetSelfSignedCert(CXPLAT_SELF_SIGN_CERT_USER, FALSE);
+    if (!CredConfig) {
+        goto CredCreateFail;
+    }
+
+    if (!QUIC_SUCCEEDED(
         MsQuic->ConfigurationOpen(
             Registration,
             Alpns,
             AlpnCount,
             &QuicSettings,
             sizeof(QuicSettings),
             nullptr,
-            &ServerConfiguration));
+            &ServerConfiguration))) {
+        goto ConfigOpenFail;
+    }
 
     ASSERT_ON_NOT(ServerConfiguration);
 
-    auto CredConfig = CxPlatGetSelfSignedCert(CXPLAT_SELF_SIGN_CERT_USER, FALSE);
-    ASSERT_ON_NOT(CredConfig);
-
-    ASSERT_ON_FAILURE(
+    if (!QUIC_SUCCEEDED(
         MsQuic->ConfigurationLoadCredential(
             ServerConfiguration,
-            CredConfig));
+            CredConfig))) {
+        goto CredLoadFail;
+    }
 
-    std::vector<HQUIC> Listeners;
     for (uint32_t i = 0; i < AlpnCount; ++i) {
         for (auto &pt : Settings.Ports) {
             HQUIC Listener;
-            ASSERT_ON_FAILURE(MsQuic->ListenerOpen(Registration, SpinQuicServerHandleListenerEvent, &Connections, &Listener));
+            if (!QUIC_SUCCEEDED(
```

Comment:
Is there a reason why we won't just re-try this until it succeeds, at least, while the failing allocator is enabled?

Reply:
The motivation here is that SpinQuic can just run another iteration when these fail.  The goal is to get coverage of these APIs failing. Retrying them until success could be a more stable solution, however.

Reply:
I'd rather fail and start over. I think it will trigger more unique code coverage. If you just keep retrying, eventually you'd always progress forward in the same way as if there wasn't failure.

Reply:
I'm going to stick with the current design. We could change to retry-until-success at a later time.